test/jmap/core/id_test.dart

```diff
@@ -0,0 +1,35 @@
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+
+void main() {
+  group('invalid id', () {
+    test('should throw argument error when the id is empty', () {
+      expect(() => Id(''), throwsArgumentError);
+    });
+
+    test('should throw argument error when the id has length more than 255', () {
+      expect(
+        () => Id('a123bde34588799922229339933933933993939393939222a123bde34'
+          '588799922229339933933933993939393939222a123be3458879992222933993393393'
+          '3993939393939222a123bde34588799922229339933933933993939393939222a123bd'
+          'e34588799922229339933933933993939393939222a123bde345887999222293399339'
+          '33933993939393939222a123bde34588799922229339933933933993939393939222a1'
+          '23bde34588799922229339933933933993939393939222'),
+        throwsArgumentError
+      );
+    });
+
+    test('should throw argument error when the id start with digit', () {
+      expect(() => Id('1be23abc'), throwsArgumentError);
+    });
+
+    test('should throw argument error when the id start with dash', () {
+      expect(() => Id('_abe23abc'), throwsArgumentError);
+    });
+
+    test('should throw argument error when the id have only digit', () {
+      expect(() => Id('123456'), throwsArgumentError);
```

Comment:
I think numbers are allowed https://jmap.io/spec-core.html#the-id-data-type

James memory implementation actively relies on them.

```
 In particular, it is wise to avoid:

    Ids starting with a dash
    Ids starting with digits
    Ids that contain only digits
    ...
```

It is not clearly forbidden then you should support it!